### ISSUE TITLE
Finetuning

### ISSUE THREAD

**Issue by ekmekovski** [xiaomi-research/gemmax#4]
Thank you for your work, it is an incredible model. I wanted to ask some tips about finetuning. I finetuned the GemmaX2-28-2B-v0.1 model for some specific domain translation (like for not to translate some domain specific terms) and it was good but I m thinking about trying to add some system like message to make model follow instructions so that I can dynamically change/ add the domain specific terms to keep them in English. Do you have any recommendation for that?
Thx in advance

**Comment by gpengzhi** [comment]
Thanks for your interest!

You can construct high-quality translation pairs with terminology demonstrations and fine-tune the GemmaX models.

**Comment by MinhNghia100402** [comment]
@gpengzhi Thanks for a great repo. I've currently tried with low-source language with about 7M data points for cpt process, including lang_1 -> en (4M), lang_2 -> en (3M), and about 200k data points for sft lang_1 -> lang_2 (and vice versa). But the result I got is not really as expected, the model has continued the conversation more instead of returning the translation result (and sometimes duplicates until reaching the limit of max_new_tokens. Can you give me some suggestions to develop better? And after doing 2 full steps cpt and sft finally evaluated on the Flores set, I got worse results than the original. Thanks!

**Comment by ekmekovski** [comment]
I finetuned the model for domain specific translation and also for some bahviour such as "Do Not Translate" via using tagging in my sft data. For some languages I also had a small decrease in BLUE score for flores but I believe it is sourced by domain adoption since BLUE is using reference texts. Maybe overfitting occured in your training? Did you check loss values? @MinhNghia100402

**Comment by ekmekovski** [comment]
For sft you can reserve some of ur finetuning data for evaluation and check how the evaluation loss also change over time, probably at some point while train loss keep decreasing the train loss starts to increasing (it also depends to ur dataset and the parameters used in training) but ı m not sure about if this is the same for cpt thare maybe other evaluation metrics for it. But try to use at least 0.1 of your sft data to use evaluation , dont forget to set the stepsize to keep checkpoints.
